src/main.ts

```diff
@@ -89,10 +90,30 @@ function setupTrayItem() {
 
 function handleDownloads() {
     mainWindow.webContents.session.on('will-download', (event, item) => {
-        const savePath = path.join(
-            app.getPath('downloads'),
-            item.getFilename()
+        item.setSavePath(
+            getUniqueSavePath(item.getFilename(), app.getPath('downloads'))
         );
-        item.setSavePath(savePath);
     });
 }
+
+function getUniqueSavePath(filename: string, directory: string): string {
+    let n = 0;
+    let exists;
+    let uniqueFileSavePath;
+    const [filenameWithoutExtension, extension] = filename.split('.');
```

Comment:
So to merge this, we just need to change this line to
```
const { name, ext } = path.parse(filename);
```

Reply:
we need to join no ? not separate ??

but i get the idea there is a path util function for it too